src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -206,11 +208,18 @@ public BitbucketCloudApiClient(boolean enableCache, int teamCacheDuration, int r
         try {
             BitbucketCloudTeam team = (BitbucketCloudTeam) getTeam();
             if (team != null) {
-                ownerId = team.getId();
+                // Templating used to set the ownerId because it parses the UUID nicely
+                this.ownerRepositoriesUrl = UriTemplate.fromTemplate(REPO_USER_TEMPLATE)
+                        .set("ownerId", team.getId())
+                        .expand();
+                this.repoUrl = this.ownerRepositoriesUrl + "/" + this.repositoryName;
             } else {
                 BitbucketCloudUser user = getUser();
                 if (user != null) {
-                    ownerId = user.getId();
+                    this.ownerRepositoriesUrl = UriTemplate.fromTemplate(REPO_USER_TEMPLATE)
+                            .set("ownerId", user.getId())
+                            .expand();
+                    this.repoUrl = this.ownerRepositoriesUrl + "/" + this.repositoryName;
```

Comment:
Noooo! please use UriTemplate, we been down this road before :laughing: 

You can just avoid setting the repo on the first one.

```java
private static final String REPO_USER_TEMPLATE = V2_API_URL + "/repositories" + "{/ownerId,repo}";
```

```java
UriTemplate template = UriTemplate.fromTemplate(REPO_USER_TEMPLATE);
this.ownerRepositoriesUrl = template.set("ownerId", user.getId()).expand();
this.repoUrl = template.set("repo", this.this.repositoryName).expand();
```

Reply:
That way you can call UriTemplate template = `UriTemplate.fromTemplate(REPO_USER_TEMPLATE);` once outside the `if else` block

Reply:
ahh okay thanks, did not realize you could do that

Reply:
Yup, templating is pretty nice :blush:

Reply:
i guess it really is..."damnhandy" :badjoke